process explorer/backend/LocalCollector/ProcessInfoCollector.cs

```diff
@@ -0,0 +1,259 @@
+﻿/* Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. */
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+using ProcessExplorer.LocalCollector.Communicator;
+using ProcessExplorer.LocalCollector.Connections;
+using ProcessExplorer.LocalCollector.EnvironmentVariables;
+using ProcessExplorer.LocalCollector.Logging;
+using ProcessExplorer.LocalCollector.Modules;
+using ProcessExplorer.LocalCollector.Registrations;
+using ProcessExplorer.LocalCollector.Connections.Interfaces;
+
+namespace ProcessExplorer.LocalCollector
+{
+    public class ProcessInfoCollector : IProcessInfoCollector
+    {
+        #region Properties
+
+        public ProcessInfoCollectorData Data { get; } = new ProcessInfoCollectorData();
+        private ICommunicator? channel;
+        private readonly ILogger<ProcessInfoCollector>? logger;
+        private readonly AssemblyInformation assemblyID = new AssemblyInformation();
+
+        private readonly object locker = new object();
+
+        #endregion
+
+        #region Constructors
+
+        ProcessInfoCollector(ICommunicator? channel = null, ILogger<ProcessInfoCollector>? logger = null,
+            string? assemblyId = null, int? pid = null)
+        {
+            this.channel = channel;
+            this.logger = logger;
+
+            if (assemblyId is not null)
+            {
+                this.assemblyID.Name = assemblyId;
+            }
+
+            if (pid is not null)
+            {
+                Data.Id = Convert.ToInt32(pid);
```

Comment:
I would think that if the type of `pid` is `int?` and it `is not null` then it must be `int`, so there should be no need to convert.

Reply:
I think we won't need neccessarily to give the pid to the ctor. 
If I use type int?, then I should convert explicitly to int, because it cannot  do that implicitly after I check.


_By default it will read the pid if it is null -in the ProcessInfoCollectorData. 
 _public int Id { get; set; } = Process.GetCurrentProcess().Id;__

Reply:
Sorry, you're right. I forgot that it's a value type, so it gets boxed, that's why it doesn't convert implicitly.